### ISSUE TITLE
Incomplete documentation for BOOT_LOADER_RAM_MEMCPY

### ISSUE THREAD

**Issue by jannic** [rp-rs/rp2040-boot2#14]
As mentioned in https://github.com/rp-rs/rp2040-boot2/pull/3#issuecomment-922084869, the documentation of BOOT_LOADER_RAM_MEMCPY seems to be incomplete.

**Comment by thejpster** [comment]
Paging @debuti, who wrote I think wrote the memcpy bootloader.

**Comment by d0ntrash** [comment]
@jannic, do you have an update on this? Do you have an working copy-to-ram example?

**Comment by jannic** [comment]
@d0ntrash no, sorry, I don't have a working example ready. I don't really have a use case for this and only tried it out last September out of curiosity. The git branch I used back then probably still is somewhere on my hard drive, so I could dig it out if it would help you. But it wouldn't be documented at all, and I'm not sure if it was correct - all I verified was that I got a LED blinking, after applying some ugly hacks.

**Comment by d0ntrash** [comment]
@jannic If that wouldn't be a big hassle for you, that would help a lot.

**Comment by sszilvasi** [comment]
@d0ntrash Have you had any luck with getting this to work?

**Comment by d0ntrash** [comment]
@sszilvasi I am not sure yet. I tried a few things but was not able to verify yet. I might have another look the next few days.

**Comment by sszilvasi** [comment]
@d0ntrash and @jannic: Do you know if there has been any `rp-rs` effort to implement the equivalent of the pico-sdk [`NO_FLASH`](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_standard_link/memmap_no_flash.ld) option?

**Comment by sszilvasi** [comment]
For the no-FLASH option, I ended up creating a [proof-of-concept RAM-only project template](https://github.com/sszilvasi/rp2040-project-template/tree/ram-only) by comparing two memory layouts in `pico-sdk`:

1. https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_standard_link/memmap_default.ld
2. https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_standard_link/memmap_no_flash.ld

As the RP Pico used for development does actually have a FLASH, I "disabled" it by flashing [picoblank](https://github.com/majbthrd/pico-demos/tree/master/picoblank) to the boot loader region.

The RAM-only project currently works with `elf2uf2-rs` but not with `probe-run`.

**Comment by sszilvasi** [comment]
Correct, the above mentioned RAM-only project was a deviation from the original topic. It is very loosely related in that the firmware gets copied to RAM by _some means_ before execution, but has otherwise little to do with the BOOT_LOADER_RAM_MEMCPY second stage boot loader implementation or its documentation.

**Comment by elfmimi** [comment]
Thanks @jannic , your [run-from-ram](https://github.com/jannic/rp2040-project-template/tree/run-from-ram) example helped much. though it didn't work as-is.

First, `.vector_table` has to be copied to RAM as well.

and there were some alignment related problems.

Finally, to convince elf2uf2 or elf2uf2-rs that your binary is meant to be placed into flash memory, you need to change entry point of the binary. That is , to change
> ENTRY(Reset);

to
> ENTRY(BOOT2_FIRMWARE);

Here is a complete working example.
https://github.com/elfmimi/rust-dap/tree/ram-lto-asm-unsafe/boards/rp2040_ram

**Comment by debuti** [comment]
Hello

Sorry to be away, you can see how it is used in this project: https://github.com/debuti/keso

BR
